ydb/core/scheme/scheme_tablecell.cpp

```diff
@@ -63,6 +63,187 @@ TOwnedCellVec::TInit TOwnedCellVec::Allocate(TOwnedCellVec::TCellVec cells) {
     };
 }
 
+namespace {
+
+    struct TCellHeader {
+        TCellHeader() = default;
+
+        TCellHeader(ui32 rawValue) : RawValue(rawValue) {}
+
+        TCellHeader(ui32 cellSize, bool isNull)
+            : RawValue(cellSize | (static_cast<ui32>(isNull) << 31ULL))
+        {}
+
+        ui32 CellSize() const { return RawValue & ~(1ULL << 31); }
+
+        bool IsNull() const { return RawValue & (1ULL << 31); };
+
+        ui32 RawValue = 0;
+    };
+
+    static_assert(sizeof(TCellHeader) == sizeof(ui32));
+
+    static constexpr ui16 CellVecSerializationHeader = -1;
+
+    Y_FORCE_INLINE void SerializeCellVec(TConstArrayRef<TCell> cells, TString &resultBuffer, TVector<TCell> *resultCells,
+                                         bool useOldSerialization = false) {
+        resultBuffer.clear();
+        if (resultCells)
+            resultCells->clear();
+
+        if (cells.empty())
+            return;
+
+        size_t size = 0;
+        if (!useOldSerialization) {
```

Comment:
I think two separate functions would be cleaner than a bunch of conditionals, and they should be available from user code, e.g. user should be able to call and test `TSerializedCellVec::Serialize_v0` vs `TSerializedCellVec::Serialize_v1`. Then constructor and default serialize could call either v0 or v1 variant.

Also there should be a test (using public apis) that serializes and deserializes both formats irrespective of current default.

Reply:
Updated.